docs/okeps/okep-5296-bgp.md

```diff
@@ -0,0 +1,663 @@
+   what routes should be received from the provider network.
+3. Create the OVN-Kubernetes RouteAdvertisements CR to configure what routes and where to advertise them.
+4. Verify peering and that routes have been propagated correctly.
+5. If desired, change the transport type from geneve on the network to none in order to no longer use an overlay.
+
+For detailed examples, see the [BGP Configuration](#bgp-configuration) section.
+
+### API Extensions
+FRR-K8S API will be used in order to create BGP Peering and configure other BGP related configuration. A
+RouteAdvertisements CRD will be introduced in order to determine which routes should be advertised for specific networks.
+Additionally, the network CRD will be modified in order to expose a new transport field to determine if encapsulation
+should be used for east/west traffic.
+
+### API Details
+
+#### Route Advertisements
+
+When OVN-Kubernetes detects that a FRR-K8S CR has been created for BGP peering, OVN-Kubernetes will by default
+advertise the pod subnet for each node, via creating an additive, per node FRR-K8S CR that is managed by OVN-Kubernetes.
+OVN-Kubernetes CRD:
+
+```yaml
+apiVersion: k8s.ovn.org/v1
+kind: RouteAdvertisements
+metadata:
+  name: default
+spec:
+    # networkSelector:
+    # nodeSelector:
+    # frrConfigurationSelector:
+    targetVRF: default
+    advertisements:
+      podNetwork: true
+      egressIP: true
+```      
+
+In the above example, an optional networkSelector may also be optionally supplied which will match namespaces/networks.
+When omitted, RouteAdvertisements will be applied to the default cluster network. If a selector is used that selects the
+default cluster network, it will be enabled for any namespace using the cluster default network. Additionally,
+advertisements may be limited to specific nodes via the nodeSelector.
+
+A networkSelector may select more than just the cluster default network, including user-defined networks. In such a case
+the network subnets will be checked by OVN-Kubernetes to determine if there is any overlap of the IP subnets. If so, an
+error status will be reported to the CRD and no BGP configuration will be done by OVN-Kubernetes.
+
+Multiple CRs may not select the same network. If this happens OVN-Kubernetes will report an error to the
+RouteAdvertisements CR and will refuse to apply the config.
+
+The CRD will support enabling advertisements for pod subnet and egress IPs. Note, MetalLB still handles advertising
+LoadBalancer IP so there is no conflict of responsibilities here. When the pod network is set to be advertised, there is
+no longer a need to SNAT pod IPs for this network to the node IP. Therefore, when pod network advertisements are enabled,
+the traffic from these pods will no longer be SNAT'ed on egress.
+
+The "targetVRF" key is used to determine which VRF the routes should be advertised in. The default value is "default", in
```

Comment:
IIUC this field is only really relevant for VRFLite or is that not true?

Reply:
it will also be relevant for EVPN